module/extension/FileWatcher/src/FileWatcher.cpp

```diff
@@ -291,22 +292,20 @@ namespace module::extension {
     }
 
     FileWatcher::~FileWatcher() {
-        for (const auto& path : paths) {
+        // Remove all the handles to the uv loop will finish
+        for (auto& path : paths) {
             if (path.second.handle) {
-                uv_fs_event_stop(path.second.handle.get());
-                uv_close(reinterpret_cast<uv_handle_t*>(path.second.handle.get()), [](uv_handle_t* /* handle */) {});
+                path.second.handle.reset();
```

Comment:
Do we need these anymore if we have the variables? they'll delete themselves anyway

Reply:
As per the comment, they need to be freed for libuv to finish. We wait for libuv to finish in the deconstructor of FileWatcher.